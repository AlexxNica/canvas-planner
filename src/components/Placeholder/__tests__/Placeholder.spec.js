import React from 'react';
import { shallow } from 'enzyme';
import 'instructure-ui/lib/themes/canvas';
import { ThemeablePlaceholder as Placeholder } from '../index';

it('renders correctly', () => {
  const wrapper = shallow(
    <Placeholder />
  )
  expect(wrapper).toMatchSnapshot();
});

it('renders the proper value', () => {
  const wrapper = shallow(
    <Placeholder value={100} />
  )
  expect(wrapper).toMatchSnapshot();
});

it('calls the onClick prop when the button is clicked', () => {
  const spy = jest.fn();
  const wrapper = shallow(
    <Placeholder onClick={spy} />
  );
  wrapper.find('Button').simulate('click');
  expect(spy).toHaveBeenCalled();
})
