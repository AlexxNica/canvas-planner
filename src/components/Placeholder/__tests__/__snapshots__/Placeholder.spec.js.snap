// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span />
  <Button
    as="button"
    buttonRef={[Function]}
    isBlock={false}
    margin="0"
    onClick={[Function]}
    size="medium"
    type="button"
    variant="default"
  >
    Click Me!
  </Button>
</div>
`;

exports[`renders the proper value 1`] = `
<div>
  <span>
    100
  </span>
  <Button
    as="button"
    buttonRef={[Function]}
    isBlock={false}
    margin="0"
    onClick={[Function]}
    size="medium"
    type="button"
    variant="default"
  >
    Click Me!
  </Button>
</div>
`;
